runtime/doc/api.txt

```diff
@@ -1350,6 +1350,27 @@ nvim_out_write({str})                                       *nvim_out_write()*
                 Parameters: ~
                     {str}  Message
 
+nvim_parse_cmdline({str}, {opts})                       *nvim_parse_cmdline()*
+                Parse command line.
+
+                Doesn't check the validity of command arguments.
+
+                Attributes: ~
+                    {fast}
+
+                Parameters: ~
+                    {str}   Command line string to parse. Cannot contain "\n".
+                    {opts}  Optional parameters. Reserved for future use.
+
+                Return: ~
+                    Dictionary containing command information, with these
+                    keys:
+                    • cmd: (string) command name.
+                    • line1: (number) starting line of command range.
+                    • line2: (number) final line of command range.
+                    • force: (boolean) whether command contains a bang.
+                    • args: (array) command arguments.
```

Comment:
Even if we don't have full capability yet, we should also include fields such as whether `%` is treated literally or expanded. The field could be called `magic` but probably it needs to be a list (not binary) because of the nonsensical variations possible with Vim commands. So `magic=[]` means "literal", `magic=['%']` means "%" is interpreted in the magic way.

This knowledge might need to be maintained as a hacky list for now, but ultimately should be built into `src/nvim/ex_cmds.lua`

Reply:
Having both `|` and `"` in magic would be redundant btw since one implies the other

Reply:
I implemented `magic` btw

Reply:
> And honestly, I don't think having | or " in magic is even necessary since we already do the job of separating the next command in nvim_parse_cmd itself

An explicit field is useful. The point is to return all knowledge about the command. Without an explicit field, the caller would have to guess.